engine/liveupdate/src/liveupdate_async.cpp

```diff
@@ -91,38 +91,38 @@ namespace dmLiveUpdate
 
         // Liveupdate async thread batch processing requested liveupdate tasks
         AsyncResourceRequest request;
-        while (m_Active)
+        while (dmAtomicGet32(&m_Active))
         {
             DM_PROFILE("Update");
             // Lock and sleep until signaled there is requests queued up
             {
                 dmMutex::ScopedLock lk(m_ConsumerThreadMutex);
                 while(m_ThreadJobQueue.Empty())
                     dmConditionVariable::Wait(m_ConsumerThreadCondition, m_ConsumerThreadMutex);
-                if((m_ThreadJobComplete) || (!m_Active))
+                if(dmAtomicGet32(&m_ThreadJobComplete) || !dmAtomicGet32(&m_Active))
                     continue;
                 request = m_ThreadJobQueue.Back();
                 m_ThreadJobQueue.Pop();
             }
             ProcessRequest(request);
-            m_ThreadJobComplete = true;
+            dmAtomicStore32(&m_ThreadJobComplete, 1);
```

Comment:
Another case in our engine code.
While a small chance, it may still produce issues if unlucky with many jobs (files)